ts/ui/menu/AnnotationMenu.ts

```diff
@@ -0,0 +1,188 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2019-2022 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  The annotations submenus.
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ * @author v.sorge@mathjax.org (Volker Sorge)
+ */
+
+import {Submenu} from 'mj-context-menu/js/item_submenu.js';
+import {MJContextMenu} from './MJContextMenu.js';
+import {MmlNode} from '../../core/MmlTree/MmlNode.js';
+import {SelectableInfo} from './SelectableInfo.js';
+import * as MenuUtil from './MenuUtil.js';
+
+/**
+ * The annotation types to look for in a MathItem. These are options set in the
+ * Menu object.
+ */
+let annotationTypes: {[type: string]: string[]} = {};
```

Comment:
The type `{[type: string]: string[]}` gets used several times.  Do you think a type declaration like

``` js
type ANNOTATIONTYPES = {[type: string]: string[]};
```

would be in order?

Reply:
Added the type.